specification/app/resource-manager/Microsoft.App/preview/2023-08-01-preview/Patches.json

```diff
@@ -0,0 +1,542 @@
+              "description": "The framework and its version in the new run image for the target image."
+            },
+            "osAndVersion": {
+              "type": "string",
+              "description": "The OS name and its version in the new run image for the target image."
+            }
+          },
+          "description": "The details of new run image for the target image.",
+          "x-nullable": true
+        },
+        "oldLayer": {
+          "type": "object",
+          "properties": {
+            "name": {
+              "type": "string",
+              "description": "The details of the old layer for the target image."
+            },
+            "frameworkAndVersion": {
+              "type": "string",
+              "description": "The framework and its version in the old run image for the target image."
+            },
+            "osAndVersion": {
+              "type": "string",
+              "description": "The OS name and its version in the old run image for the target image."
+            }
+          },
+          "description": "The details of old run image for the target image.",
+          "x-nullable": true
+        },
+        "patchType": {
+          "type": "string",
+          "description": "The type for the patch.",
+          "enum": [
+            "FrameworkSecurity",
+            "OSSecurity",
+            "FrameworkAndOS",
+            "Other"
+          ],
+          "x-nullable": true,
+          "x-ms-enum": {
+            "name": "PatchType",
+            "modelAsString": true
+          }
+        }
+      },
+      "readOnly": true
+    },
+    "PatchProperties": {
+      "type": "object",
+      "properties": {
+        "targetEnvironmentId": {
+          "format": "arm-id",
+          "description": "The id of the target environment for the patch."
+        },
+        "targetContainerAppId": {
+          "format": "arm-id",
+          "description": "The id of the target container app for the patch."
+        },
+        "targetRevisionId": {
+          "format": "arm-id",
+          "description": "The id of the target revision for the patch."
+        },
+        "provisioningState": {
+          "$ref": "#/definitions/PatchProvisioningState",
+          "description": "Patch provisioning state.",
+          "readOnly": true
+        },
+        "patchApplyStatus": {
+          "$ref": "#/definitions/PatchApplyStatus",
+          "description": "The status of the patch once it has been provisioned",
+          "readOnly": true
+        },
+        "createdTime": {
+          "type": "string",
+          "format": "date-time",
+          "description": "The UTC timestamp that describes when the patch object was created."
+        },
+        "updatedTime": {
+          "type": "string",
+          "format": "date-time",
+          "description": "The UTC timestamp that describes when the patch object was last updated."
+        },
+        "patchDetails": {
```

Comment:
>patchDetails [](http://example.com/codeflow?start=9&length=12)

I found this a bit confusing. A patch has properties, which has details. Why the need or two levels? Is there a better subcategory name you could give the 'details'?

Reply:
yes, so the patch details could list an array of patch operation that is going to perform when user applies the patch by its patch name. The patch properties are just the properties needed to describe the entire patch (e.g. the patching status, etc.)